File: app/src/main/java/com/tcc/sicv/data/preferences/PreferencesHelper.java
package com.tcc.sicv.data.preferences;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

import static com.tcc.sicv.utils.Constants.PREF_KEY;
import static com.tcc.sicv.utils.Constants.USER_FIELD;

public class PreferencesHelper {
    private SharedPreferences sp;

    public PreferencesHelper(Application applicationContext) {
        sp = applicationContext.getSharedPreferences(PREF_KEY , Context.MODE_PRIVATE);
    }

    public void saveUser(String userEmail){
        SharedPreferences.Editor editor = sp.edit();
        editor.putString(USER_FIELD, userEmail);
        editor.apply();
    }

    public String getEmail(){
        if(!sp.contains(USER_FIELD)) return null;
        return sp.getString(USER_FIELD, "");
    }

    public void logout(){
        if(!sp.contains(USER_FIELD)) return;
        SharedPreferences.Editor editor = sp.edit();
        editor.remove(USER_FIELD);
        editor.apply();

    }
}

File: app/src/main/java/com/tcc/sicv/data/preferences/PreferencesHelper.kt
package com.tcc.sicv.data.preferences

import android.app.Application
import android.content.Context
import android.content.SharedPreferences

import com.tcc.sicv.utils.Constants.PREF_KEY
import com.tcc.sicv.utils.Constants.USER_FIELD

class PreferencesHelper(applicationContext: Application) {
    private val sp: SharedPreferences = applicationContext.getSharedPreferences(PREF_KEY, Context.MODE_PRIVATE)

    val email: String?
        get() = if (!sp.contains(USER_FIELD)) null else sp.getString(USER_FIELD, "")

    fun saveUser(userEmail: String) {
        val editor = sp.edit()
        editor.putString(USER_FIELD, userEmail)
        editor.apply()
    }

    fun logout() {
        if (!sp.contains(USER_FIELD)) return
        val editor = sp.edit()
        editor.remove(USER_FIELD)
        editor.apply()

    }
}
